uwblueprint/starter-code-cli
# Add initial CLI which prompts user for starter code options

## Notion ticket link

[POC of CLI (single option)](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d?p=cd946ac38a6a4860b1898b665264c7bc)



## Implementation description
* Used inquirer package for prompts
* Used boxes for box, chalk for coloured prompts
* Used yargs to handle command line options
* Used figlet for ASCII art



## Steps to test
1. Compile with tsc -p . and run node bin/index.js. Verify that the confirmation prompt matches the user's options.
<img width="1036" alt="image" src="https://user-images.githubusercontent.com/37782734/109252454-935f0600-77bb-11eb-91d0-6fdc8e749cba.png">
<img width="980" alt="image" src="https://user-images.githubusercontent.com/37782734/109252475-9e199b00-77bb-11eb-8215-dd2ab1398c75.png">
2. Run node bin/index.js with the command line options seen with node index.js help. Verify that the CLI prompts the user for missing options and that the confirmation prompt matches the user's options
<img width="980" alt="image" src="https://user-images.githubusercontent.com/37782734/109252503-b4bff200-77bb-11eb-89ff-e814fe1d2e76.png">




## What should reviewers focus on?
* Have not worked a lot with Typescript on Node, so it would be helpful to know if anything needs to be added for best practices


## Checklist
